syntax = "proto2";

message RecordSegmentSetting {
  // Segmentation of restored records in size or time, default is by size 200MB
  optional int32 size_segment = 1 [default = 200];
  optional int32 time_segment = 2;
}

message Trigger {
  optional string trigger_name = 1;
  optional bool enabled = 2;
  optional double backward_time = 3;
  optional double forward_time = 4;

  // Messages we need to capture when trigger is set
  // It's a map of channel names and message types
  map<string, string> wanted_channel_types = 5;
}

message SmartRecordTrigger {
  optional RecordSegmentSetting segment_setting = 1;
  repeated Trigger triggers = 2;
}
